parachain/pallets/ethereum-beacon-light-client/src/lib.rs

```diff
@@ -0,0 +1,710 @@
+pub struct BeaconBlockHeader {
+	// The slot for which this block is created. Must be greater than the slot of the block defined by parentRoot.
+	pub slot: u64,
+	// The index of the validator that proposed the block.
+	pub proposer_index: ValidatorIndex,
+	// The block root of the parent block, forming a block chain.
+	pub parent_root: Root,
+	// The hash root of the post state of running the state transition through this block.
+	pub state_root: Root,
+	// The hash root of the Eth1 block
+	pub body_root: Root,
+}
+
+/// Sync committee as it is stored in the runtime storage.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SyncCommittee {
+	pub pubkeys: Vec<PublicKey>,
+	pub aggregate_pubkey: PublicKey,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SyncAggregate {
+	// 1 or 0 bit, indicates whether a sync committee participated in a vote
+	pub sync_committee_bits: Vec<u8>,
+	pub sync_committee_signature: Vec<u8>,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct LightClientInitialSync {
+	pub header: BeaconBlockHeader,
+	pub current_sync_committee: SyncCommittee,
+	pub current_sync_committee_branch: ProofBranch,
+	pub validators_root: Root,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct LightClientSyncCommitteePeriodUpdate {
+	pub attested_header: BeaconBlockHeader,
+	pub next_sync_committee: SyncCommittee,
+	pub next_sync_committee_branch: ProofBranch,
+	pub finalized_header: BeaconBlockHeader,
+	pub finality_branch: ProofBranch,
+	pub sync_aggregate: SyncAggregate,
+	pub fork_version: ForkVersion,
+	pub sync_committee_period: u64,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct LightClientFinalizedHeaderUpdate {
+	pub attested_header: BeaconBlockHeader,
+	pub finalized_header: BeaconBlockHeader,
+	pub finality_branch: ProofBranch,
+	pub sync_aggregate: SyncAggregate,
+	pub fork_version: ForkVersion,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct ForkData {
+	// 1 or 0 bit, indicates whether a sync committee participated in a vote
+	pub current_version: [u8; 4],
+	pub genesis_validators_root: [u8; 32],
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SigningData {
+	pub object_root: Root,
+	pub domain: Domain,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct Genesis {
+	pub validators_root: Root,
+}
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use super::*;
+
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	use milagro_bls::{AggregatePublicKey, AggregateSignature, AmclError, Signature};
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
```

Comment:
```suggestion
```

It shouldn't be necessary to use this macro, as long as all storage items implement [MaxEncodedLen](https://crates.parity.io/frame_support/pallet_prelude/trait.MaxEncodedLen.html). Which should already be the case from what I see. If not, lets update those types. Can check to see whether this is necessary by trying to compile the pallet after removing the macro.

There is currently an issue in our tracker to remove all uses of `without_storage_info`: https://linear.app/snowfork/issue/SNO-223/implement-maxencodedlen-trait-for-pallets

Reply:
As discussed, I will do this in a follow-up PR since I need to replace the occurrences stored in `Vec` with `BoundedVec`.

Reply:
Logged Linear ticket [SNO-264](https://linear.app/snowfork/issue/SNO-264/implement-maxencodedlen-trait-for-beacon-client) for this.